noir-projects/noir-contracts/contracts/nft_contract/src/main.nr

```diff
@@ -0,0 +1,191 @@
+// Minimal NFT implementation with `AuthWit` support that allows minting in public-only and transfers in both public
+// and private.
+contract NFT {
+    #[aztec(event)]
+    struct Transfer {
+        from: AztecAddress,
+        to: AztecAddress,
+        token_id: Field,
+    }
+
+    #[aztec(storage)]
+    struct Storage {
+        // The symbol of the NFT
+        symbol: SharedImmutable<FieldCompressedString>,
+        // The name of the NFT
+        name: SharedImmutable<FieldCompressedString>,
+        // The admin of the contract
+        admin: PublicMutable<AztecAddress>,
+        // Addresses that can mint
+        minters: Map<AztecAddress, PublicMutable<bool>>,
+        // Contains the NFTs owned by each address in private.
+        private_nfts: Map<AztecAddress, PrivateSet<NFTNote>>
+        // Contains number of NFTs owned by each address in public.
+        public_balances: Map<AztecAddress, PublicMutable<U128>>,
+        // A map from token ID to a boolean indicating if the NFT exists.
+        nft_exists: Map<Field, PublicMutable<bool>>,
+        // A map from token ID to the public owner of the NFT.
+        public_owners: Map<Field, PublicMutable<AztecAddress>>,
```

Comment:
> I'm skeptical re. how useful this would be tbh - I think the questions Rahul is asking re. proving ownership are the way to go.

The public_owners map is essential to tracking who owns a given NFT in public. Are you proposing with this nuking of the public state here and just having private NFTs?